chainer/initializer.py

```diff
@@ -3,6 +3,9 @@
 
 class Initializer(object):
 
+    def __init__(self, dtype=None):
```

Comment:
Please specify what is expected to `dtype` argument here. Unless I overlook something, it is for type check of `__call__` functions and `generate_array` function.